### PR TITLE
use anr pid in e2e test yaml

### PR DESCRIPTION
**Pull request by felipemadero**
Currently e2e test binary is writing its own pid in the yaml it generates with ANR info. This PR changes that so that the yaml contains ANR server pid. 
Also changes simulator flag name so the yaml is not interpreted as ANR-generated.

### CODE REVIEW

**Review by darioush — APPROVED** [review]

**Comment by darioush** [comment]
can you add a description to the pr?

File: tests/e2e/runner/runner.go [hunk 1]
```diff
@@ -136,20 +140,23 @@ done:
...
+	}
+	if pid == 0 {
+		return "", "", pid, errors.New("pid not found")
 	}
-	return blockchainID, logsDir, nil
+	return blockchainID, logsDir, pid, nil
 }
 
-func GetClusterInfo(blockchainId string, logsDir string) (clusterInfo, error) {
+func GetClusterInfo(blockchainId string, logsDir string, pid int) (clusterInfo, error) {
```
Comment (aaronbuchwald) [inline tests/e2e/runner/runner.go:159]: We should not need to pass a process ID into a getter function to return in the output.

File: tests/e2e/runner/runner.go [hunk 2]
```diff
@@ -92,8 +92,9 @@ func startRunner(vmName string, genesisPath string, pluginDir string) error {
 	return nil
 }
 
-func WaitForCustomVm(vmId ids.ID) (string, string, error) {
+func WaitForCustomVm(vmId ids.ID) (string, string, int, error) {
```
Comment (aaronbuchwald) [inline tests/e2e/runner/runner.go:95]: Can we move this logic inside of the avalanche-network-runner? I would think that we can move the for loop here into the avalanche-network-runner and simplify this code which will likely need to be duplicated in any project doing this similar and most likely very common task.

**Review by aaronbuchwald — APPROVED** [review]

File: tests/e2e/runner/runner.go (hunk 1 above)
Comment (felipemadero) [reply tests/e2e/runner/runner.go:159]: both logsDir and pid are passed to this probably ill-named function that also ask server for uris and save cluster info data (including given logsDir+pid) to disk before returning it. Renaming to SaveClusterInfo

File: tests/e2e/runner/runner.go (hunk 2 above)
Comment (felipemadero) [reply tests/e2e/runner/runner.go:95]: This is already included for second ANR refactoring, but will anyway make a explicit issue now and work first on it. https://github.com/ava-labs/avalanche-network-runner/issues/302
After doing that, will add issues to modify CLI/subnet-evm

**Pull request merged**